app/controllers/api/v2/sessions_controller.rb

```diff
@@ -21,6 +21,31 @@ def create
         render jsonapi: @current_user
       end
 
+      def create_by_magic
```

Comment:
It would be good to extract this into a service and/or `MagicLinkToken` class. Seems like a lot of steps and also some important logic that it would be good to have unit test coverage of.

Reply:
Another case of expediency / laziness. Can we do that as part of another ticket? Like the one about not returning an error if the user doesn't exist, which will touch on this bit of code.

Reply:
👍 https://trello.com/c/vHEoXZQK/3359-m-add-magic-link-journey-for-non-existent-users